Automattic/jetpack
# Social Menu: add regex option to service detection in menu.

#### Changes proposed in this Pull Request:

Up until now, social menu items would only rely on host names (e.g. amazon.fr) to detect whether a specific menu item should use a specific icon.
This works well, but isn't very flexible. It's especially problematic for services that typically either have lots of different TLDs, or services that use different domain names.

With this commit, you should now be able to provide either a host name, just like before and ensuring backwards compatibility for folks who have customized that feature on their end, as well as regex for more advanced matching.

In future PRs, I'm hoping to use this new flexibility to add support for more services.

#### Other information:


#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Start by switching to an old theme on your site, like Twenty Ten.
* In a functionality plugin, add the following code snippet:
```php
// Add social menu support
add_action( 'after_setup_theme', function () {
	add_theme_support( 'jetpack-social-menu', 'svg' );
} );
```
* Go to Appearance > Menus, and create a new menu.
* In that menu, add a few custom links pointing to different services supported by Jetpack: Twitter, Strava, ..., and an Amazon link of your choice. Also add an unsupported service, like a link to `jetpack.com` for example.
* After saving your menu, assign it to the Social Menu location.
* Save your changes.
* Visit the site's frontend and check the look of the menu.
